### PR TITLE
Lines with permanent current limits

### PR DESCRIPTION
**Pull request by sBouzols**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/dto/CurrentLimitsInfos.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@SuperBuilder
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Schema(description = "Current Limits")
+public class CurrentLimitsInfos {
+
+    @Schema(description = "Permanent current limit")
+    private Double permanentCurrentLimit;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/dto/CurrentLimitsInfos.java:29]: permanentLimit ? (no use to repeat current)

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import javax.persistence.Table;
+
+/**
+ * @author Sylvain Bouzols <sylvain.bouzols at rte-france.com>
+ */
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "currentLimit")
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java:31]: currentLimits

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id")
+    private UUID id;
+
+    @Column(name = "permanentCurrentLimit")
+    private Double permanentCurrentLimit;
+
+    // public CurrentLimitEntity(CurrentLimitEntity that) {
+    //     this.permanentCurrentLimit = that.permanentCurrentLimit;
+    // }
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java:44]: to remove

File: src/test/java/org/gridsuite/modification/server/utils/MatcherLineCreationInfos.java [hunk 4]
```diff
@@ -60,6 +65,12 @@ protected MatcherLineCreationInfos(LineCreationInfos ref) {
         super(ref);
     }
 
+    public boolean matchesCurrentLimitsInfos(CurrentLimitsInfos cl1, CurrentLimitsInfos cl2) {
+        return cl1 == null && cl2 == null
+            || cl1 != null && cl2 != null && cl1.getPermanentCurrentLimit().equals(cl2.getPermanentCurrentLimit())
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/utils/MatcherLineCreationInfos.java:70]: Pb if cl1.getPermanentCurrentLimit() is null
Add parenthesis ?

File: src/test/java/org/gridsuite/modification/server/ModificationRepositoryTest.java [hunk 5]
```diff
@@ -254,13 +254,13 @@ public void testGeneratorCreation() {
...
-        var createLineEntity3 = modificationRepository.createLineEntity("idLine3", "nameLine3", 3.0, 3.3, 30.0, 33.0, 300.0, 300.3, "vlId31", "busId31", "vlId32", "busId32");
-        var createLineEntity4 = modificationRepository.createLineEntity("idLine4", "nameLine4", 3.0, 3.3, null, null, null, null, "vlId41", "busId41", "vlId42", "busId42");
+        var createLineEntity1 = modificationRepository.createLineEntity("idLine1", "nameLine1", 1.0, 1.1, 10.0, 11.0, 100.0, 100.1, "vlId11", "busId11", "vlId12", "busId12", null, null);
+        var createLineEntity2 = modificationRepository.createLineEntity("idLine2", "nameLine2", 2.0, 2.2, 20.0, 22.0, 200.0, 200.2, "vlId21", "busId21", "vlId22", "busId22", null, null);
+        var createLineEntity3 = modificationRepository.createLineEntity("idLine3", "nameLine3", 3.0, 3.3, 30.0, 33.0, 300.0, 300.3, "vlId31", "busId31", "vlId32", "busId32", null, 5.0);
+        var createLineEntity4 = modificationRepository.createLineEntity("idLine4", "nameLine4", 3.0, 3.3, null, null, null, null, "vlId41", "busId41", "vlId42", "busId42", 5.0, null);
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/modification/server/ModificationRepositoryTest.java:260]: Add test with permanent limits on both sides

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/dto/CurrentLimitsInfos.java (hunk 1 above)
Comment (sBouzols) [reply src/main/java/org/gridsuite/modification/server/dto/CurrentLimitsInfos.java:29]: ok

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java (hunk 2 above)
Comment (sBouzols) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java:31]: ok + changeset then

File: src/test/java/org/gridsuite/modification/server/ModificationRepositoryTest.java (hunk 5 above)
Comment (sBouzols) [reply src/test/java/org/gridsuite/modification/server/ModificationRepositoryTest.java:260]: ok

File: src/test/java/org/gridsuite/modification/server/utils/MatcherLineCreationInfos.java (hunk 4 above)
Comment (sBouzols) [reply src/test/java/org/gridsuite/modification/server/utils/MatcherLineCreationInfos.java:70]: ok

File: src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java (hunk 3 above)
Comment (sBouzols) [reply src/main/java/org/gridsuite/modification/server/entities/equipment/creation/CurrentLimitsEntity.java:44]: ok

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**